### PR TITLE
Fix for WFCORE-4406 and Upgrade to Galleon 4.0.0.alpha

### PR DESCRIPTION
**Pull request by jfdenise**
F- ix for https://issues.jboss.org/browse/WFCORE-4406
- Upgrade to galleon 4.0 alpha. Final upgrade is tracked by (https://issues.jboss.org/browse/WFCORE-4422)
- Produces FP with Galleon 4.0, generates and deploy wildfly-core offliner. That is a required step to address: https://issues.jboss.org/browse/WFLY-11989
- Make dependencies provided, it is required to not download all dependencies when building User feature-packs from wildfly feature-packs.

### CODE REVIEW

**Comment by kabir** [comment]
@psakar @bstansberry FYI this is not labelled with deps-changed as it does not go into the feature packs

File: pom.xml [hunk 1]
```diff
@@ -113,10 +113,10 @@
 
         <!-- Non-default maven plugin versions and configuration -->
         <version.jacoco.plugin>0.8.0</version.jacoco.plugin>
-        <version.org.jboss.galleon>3.0.1.Final</version.org.jboss.galleon>
+        <version.org.jboss.galleon>4.0.0.Alpha1</version.org.jboss.galleon>
         <version.org.wildfly.build-tools>1.2.10.Final</version.org.wildfly.build-tools>
         <version.org.wildfly.checkstyle-config>1.0.5.Final</version.org.wildfly.checkstyle-config>
-        <version.org.wildfly.galleon-plugins>3.0.1.Final</version.org.wildfly.galleon-plugins>
+        <version.org.wildfly.galleon-plugins>4.0.0.Alpha4</version.org.wildfly.galleon-plugins>
```
Comment (jmesnil) [inline pom.xml]: Was it a coincidence that both org.jboss.galleon & org.wildfly.galleon-plugins were at 3.0.1.Final before?
Is there any relation between the 2 versions?

Comment (jfdenise) [reply pom.xml]: @jmesnil , the difference in version is due to some issues found in galleon-plugins. The 2 galleon deps will be 4.0.Final

**Comment by jfdenise** [comment]
@jmesnil , I think that it should be ok, @aloubyansky we have 2 weeks before the final release, is it fine with you?

**Comment by aloubyansky** [comment]
Yes, we'll have the final release by that date.

**Pull request merged**